bundles/org.openhab.binding.icalendar/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -99,4 +99,113 @@
 		<description>End of the next event in calendar</description>
 		<state readOnly="true"/>
 	</channel-type>
+
+	<channel-type id="result_start">
+		<item-type>DateTime</item-type>
+		<label>Start of Result</label>
+		<description>Start of the found result in calendar</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="result_end">
+		<item-type>DateTime</item-type>
+		<label>End of Result</label>
+		<description>End of the found result in calendar</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="result_title">
+		<item-type>String</item-type>
+		<label>Title of Result</label>
+		<description>Title of the found result in calendar</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-group-type id="result">
+		<label>Result Event</label>
+		<description>A resulting event found by filter</description>
+		<channels>
+			<channel typeId="result_start" id="begin"/>
+			<channel typeId="result_end" id="end"/>
+			<channel typeId="result_title" id="title"/>
+		</channels>
+	</channel-group-type>
+
+
+	<thing-type id="eventfilter">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="calendar"/>
+		</supported-bridge-type-refs>
+		<label>Event Filter</label>
+		<description>Filtered Events from the calendar</description>
+
+		<config-description>
+			<parameter-group name="general">
+				<label>General Filter Options</label>
+			</parameter-group>
+			<parameter-group name="datetime_based">
+				<label>Date and Time based Filter</label>
+			</parameter-group>
+			<parameter-group name="text_based">
+				<label>Text based Filter</label>
+			</parameter-group>
+
+			<parameter name="maxEvents" type="integer" min="0" groupName="general">
+				<label>Maximum Matches</label>
+				<required>true</required>
+			</parameter>
+			<parameter name="refreshTime" type="integer" min="1" groupName="general" unit="min">
+				<label>Refresh Time</label>
+				<description>The frequency in minutes the channels get refreshed</description>
+				<required>true</required>
+				<default>15</default>
+			</parameter>
+			<parameter name="datetimeUnit" type="text" groupName="datetime_based">
+				<limitToOptions>true</limitToOptions>
+				<options>
+					<option value="MINUTE">minute</option>
+					<option value="HOUR">hour</option>
+					<option value="DAY">day</option>
+					<option value="WEEK">week</option>
+				</options>
+				<default>hour</default>
```

Comment:
Did you test the default value in the UI? I guess this should be "HOUR".

Reply:
Yes, you are right, i've changed the constant after the last test. I'll fix it.